### ISSUE TITLE
Sometimes process can fail to restart

### ISSUE THREAD

**Issue by pfhayes** [kmagiera/babel-watch#37]
I've noticed that when many restarts are triggered in quick succession (for example, by running a `git pull`), then an error can come up and the watch process will stop monitoring.

```
Error: Cannot find module './express/app'
    at Function.Module._resolveFilename (module.js:455:15)
    at Function.Module._load (module.js:403:25)
    at Module.require (module.js:483:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/Users/patrick/project/js/config/main.js:7:17)
    at Module._compile (module.js:556:32)
    at babelWatchLoader (/Users/patrick/project/node_modules/babel-watch/runner.js:50:13)
    at Object.require.extensions.(anonymous function) [as .js] (/Users/patrick/project/node_modules/babel-watch/runner.js:61:7)
    at Module.load (module.js:473:32)
    at tryModuleLoad (module.js:432:12)
fs.js:786
    return binding.writeBuffer(fd, buffer, offset, length, position);
                   ^

Error: EPIPE: broken pipe, write
    at Error (native)
    at Object.fs.writeSync (fs.js:786:20)
    at handleFileLoad (/Users/patrick/project/node_modules/babel-watch/babel-watch.js:279:30)
    at handleFileLoad (/Users/patrick/project/node_modules/babel-watch/babel-watch.js:172:7)
    at ChildProcess.app.on (/Users/patrick/project/node_modules/babel-watch/babel-watch.js:273:5)
    at emitTwo (events.js:106:13)
    at ChildProcess.emit (events.js:191:7)
    at process.nextTick (internal/child_process.js:744:12)
    at _combinedTickCallback (internal/process/next_tick.js:67:7)
    at process._tickCallback (internal/process/next_tick.js:98:9)
```

Normally there is no problem with importing `./express/app` - it appears to just be a transient error while the filesystem is changing. However, after this error comes up I need to kill babel-watch and restart it in order for monitoring to continue

**Comment by pfhayes** [comment]
I've made two draft PRs that address this issue. This fixes the issue for me but may be suboptimal - happy to rethink the solution or address any concerns

https://github.com/kmagiera/babel-watch/pull/38
https://github.com/kmagiera/babel-watch/pull/39

**Comment by pfhayes** [comment]
As far as I can tell, this issue has been resolved by https://github.com/kmagiera/babel-watch/pull/44

**Comment by pfhayes** [comment]
I've noticed that this issue still appears for me in the latest version

**Comment by cromwellryan** [comment]
This issue is easily reproduced in our larger project by `touch`ing two files being watched:

```
touch <fileA> <fileB>
```

In a smaller example, that same touch only results in a single `<<<RESTARTING>>>` event as would be expected. During a the `touch` scenario in our larger project, 2 consistently occur at the same `require` statement. Any given `git checkout other-branch-here` we see numerous restarts. 

I'd be happy to pair on a debugging session.

In #39, @STRML [suggests a debounce](https://github.com/kmagiera/babel-watch/pull/39#issuecomment-272269397). I've confirm that this will resolve the problem for us. 

PR #70 is an implementation of this debounce which is what we are using for the time being.

I also would have expected implementing chokidar's `awaitWriteFinish` option to resolve the issue. Alas, it did not.

**Comment by STRML** [comment]
This was released in `7.1.0`, so I'm closing this.
